correct how to install Docker SDK for Python

daeseokyoun opened the pull request:
##### SUMMARY
The latest version 5.0.0 do not support Python 2.7 anymore.

Documents and error message to guide docker/docker-py module with pip corresponding Python version.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docker

##### ADDITIONAL INFORMATION
When install  Docker SDK for Python 5.0.0 in system used Python 2.7, ansible failed to run related to docker module with following message.
```json
"msg": "Failed to import the required Python library (Docker SDK for Python: docker (Python >= 2.7) or docker-py (Python 2.6)) on cstgnmss0022.lip2's Python /usr/bin/python. Please read module documentation and install in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter, for example via `pip install docker` or `pip install docker-py` (Python 2.6). The error was: cannot import name credentials",
```
But it is not correct, the 5.0.0 do not support Python 2.7. We should install docker sdk version before 5.0.0 or install Python 3.6 in target system. so correct message following below:
```json
"msg": "Failed to import the required Python library (Docker SDK for Python: docker above 5.0.0 (Python >= 3.6) or docker before 5.0.0 (Python >= 2.7) or docker-py (Python 2.6)) on sstgnmss0003.lip2's Python /usr/bin/python. Please read module documentation and install in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter, for example via `pip install docker` (Python >= 3.6) or `pip install docker==4.4.4` (Python >= 2.7) or `pip install docker-py` (Python 2.6). The error was: cannot import name credentials"
```

jborean93 commented:
cc @felixfontein not sure if you are still working on the Docker modules, if so are you able to look at the doc changes here. As for the modification of the help code I'm not 100% sure it's needed. If the module for Python 2.7 only supports up to a certain version then `pip` should be handling that automatically without the user having to supply `docker<=4.4.4`.

felixfontein commented:
> cc @felixfontein not sure if you are still working on the Docker modules,

I still do.

> if so are you able to look at the doc changes here. As for the modification of the help code I'm not 100% sure it's needed. If the module for Python 2.7 only supports up to a certain version then `pip` should be handling that automatically without the user having to supply `docker<=4.4.4`.

I also said this before, but it looks like that doesn't really work. In community.docker's CI some jobs were failing when docker-py 5.0.0 was released (CentOS 7 and Ubuntu 16.04) - see the first commit in ansible-collections/community.docker#120. Maybe the pip versions in these images are too old to support that?

felixfontein commented on test/support/integration/plugins/module_utils/docker/common.py:
```diff
@@ -334,11 +334,13 @@ def __init__(self, argument_spec=None, supports_check_mode=False, mutually_exclu
         if not HAS_DOCKER_PY:
             if NEEDS_DOCKER_PY2:
-                msg = missing_required_lib("Docker SDK for Python: docker")
-                msg = msg + ", for example via `pip install docker`. The error was: %s"
```
Changing this file makes no sense, this is a copy that is only used in ansible/ansible's integration tests. You should modify https://github.com/ansible-collections/community.docker/blob/main/plugins/module_utils/common.py#L264 instead.

samdoran replied:
Jinx 😉

samdoran commented:
Files in `test/support/` are used only for testing and we only update them to address failures in CI. If a change needs to be made to the actual module, a PR should be submitted for the https://github.com/ansible-collections/community.docker repository since that module now lives in a collection outside this repository.

jborean93 commented:
Thanks @felixfontein for confirming, disappointing that the metadata isn't present to have `pip` automatically use the older version.

felixfontein commented:
@jborean93 the setup.py included in https://files.pythonhosted.org/packages/fa/a2/e46d7c1b51394a09271a3b07c3a68deb3a669429beafd444d9553ed52868/docker-5.0.0.tar.gz does say `python_requires='>=3.6'`, but I'm not very familiar with how setup.py files work to be able to say whether that's the correct way to do it or not...

jborean93 commented:
Maybe it requires a `pip` version not present in CI as that's how it should work.

daeseokyoun commented:
@felixfontein @jborean93 Do I need to create new PR for this to community.docker repo? how about the docs/docsite/rst/scenario_guides/guide_docker.rst file?

felixfontein commented:
@daeseokyoun you still need to change docs/docsite/rst/scenario_guides/guide_docker.rst in this repo, but not the other file. For communit.docker you need to create a new PR.

daeseokyoun commented:
@felixfontein Thanks for your comment. I will update this PR to leave only the commit related to guide_docker.rst, and create a new PR for the common.py on community.docker repo.

Commit:
docs: correct guide for the latest Docker SDK for Python

The latest version 5.0.0 do not support Python 2.7 anymore.
It need to guide how to install docker / docker-py module for
proper Python version.

felixfontein commented on docs/docsite/rst/scenario_guides/guide_docker.rst:
```diff
@@ -15,13 +15,19 @@ Most of the modules and plugins in community.docker require the `Docker SDK for
 Note that plugins (inventory plugins and connection plugins) are always executed in the context of Ansible itself. If you use a plugin that requires the Docker SDK for Python, you need to install it on the machine running ``ansible`` or ``ansible-playbook`` and for the same Python interpreter used by Ansible. To see which Python is used, run ``ansible --version``.
 
-You can install the Docker SDK for Python for Python 2.7 or Python 3 as follows:
+You can install the Docker SDK for Python for Python>=3.6 as follows:
```
```suggestion
You can install the Docker SDK for Python for Python 3.6 or later as follows:
```

daeseokyoun replied:
ok, I will update commit.

daeseokyoun committed:
indicate python version range with text

indicate available python version range with only text instead of using mathematical symbol.

Co-authored-by: Felix Fontein <felix@fontein.de>

samccann commented on docs/docsite/rst/scenario_guides/guide_docker.rst:
```diff
@@ -15,13 +15,19 @@ Most of the modules and plugins in community.docker require the `Docker SDK for
     $ pip install docker
 
-For Python 2.6, you need a version before 2.0. For these versions, the SDK was called ``docker-py``, so you need to install it as follows:
+For Python 2.7, you need to use a version between 2.0.0 and 4.4.4 since removed support for Python 2.7 on 5.0.0. You can install the specific version of the Docker SDK for Python as follows:
```
```suggestion
For Python 2.7, you need to use a version between 2.0.0 and 4.4.4 since the Python package for Docker removed support for Python 2.7 on 5.0.0. You can install the specific version of the Docker SDK for Python as follows:
```

acozine committed:
Update docs/docsite/rst/scenario_guides/guide_docker.rst

Co-authored-by: Sandra McCann <samccann@redhat.com>

acozine commented:
Thanks @daeseokyoun for this PR!
